sdk/messaging/azeventgrid/client.go

```diff
@@ -324,3 +383,65 @@ func (client *Client) releaseCloudEventsHandleResponse(resp *http.Response) (Rel
 	}
 	return result, nil
 }
+
+// RenewCloudEventLocks - Renew lock for batch of Cloud Events. The server responds with an HTTP 200 status code if the request
+// is successfully accepted. The response body will include the set of successfully renewed
+// lockTokens, along with other failed lockTokens with their corresponding error information.
+// If the operation fails it returns an *azcore.ResponseError type.
+//
+// Generated from API version 2023-10-01-preview
+//   - topicName - Topic Name.
+//   - eventSubscriptionName - Event Subscription Name.
+//   - renewLockOptions - RenewLockOptions
+//   - options - RenewCloudEventLocksOptions contains the optional parameters for the Client.RenewCloudEventLocks method.
+func (client *Client) RenewCloudEventLocks(ctx context.Context, topicName string, eventSubscriptionName string, renewLockOptions RenewLockOptions, options *RenewCloudEventLocksOptions) (RenewCloudEventLocksResponse, error) {
```

Comment:
This looks a little weird, the two options types. Is `RenewLockOptions` actually optional?

Reply:
This is all over their API at the moment, which is unfortunate.

What is our naming convention for "group of non-optional parameters"?

Reply:
Body?

Reply:
We don't have a standard prefix/suffix for such types as we should try to come up with clear, domain-specific names, thus will be on a case-by-case basis. Maybe something like `LockRenewal`, `LockRenewalValues`?

CC @JeffreyRichter

Reply:
One could also make the argument that the slice of lock tokens should be a discrete parameter.

Reply:
> One could also make the argument that the slice of lock tokens should be a discrete parameter.

I'd be a bit worried that expansion of that 'args' bag will mean I'll need to start creating multiple functions. So we should probably keep it in a bag for now, even if we need to come up with a name.

Reply:
I'm trying to get this out so we can properly deserialize all the previous data since that recently broke. 

I'll create an APIView, we can hash it out there and I'll fix it up before the next beta goes out.

Reply:
> creating multiple functions

Can you expand on that? Will the bag contain a mix of optional and/or mutually exclusive values?

Reply:
https://apiview.dev/Assemblies/Review/05f3306f2ec84f1bbdd581445bf015d6?diffRevisionId=56aa37b5b45a48e8a8b6c0dc3f0596a4&diffOnly=False&revisionId=c2c10b9a596d46e39a6df83dcde04346&doc=False

Reply:
> > creating multiple functions
> 
> Can you expand on that? Will the bag contain a mix of optional and/or mutually exclusive values?

I don't think it'll be a mix - ie, optional should be in Options, required args in "args/body/params". 

What I mean is that if they add even one more required parameter then we'd have to create multiple functions in the future. I'd rather not head down the path of assuming it's always going to be just the one required arg.

Reply:
Thinking about this some more...they can't add another required parameter. I'll expand this out for now.

Reply:
Well, if the body ends up requiring additional content, then that would be a breaking change. However, conceptually, I think that's desired as we're now saying that the contract for this method has changed in a meaningful way.

Reply:
(talked offline)

We'll talk about this some more. In this case (and the others) we can just make the one field in the bag a required parameter, which makes the entire thing nicer anyways. 

But it'd be good to just head these kinds of conversations off at the pass, pick a consistent suffix (so customers can just expect it, like we have with our normal `Options` bags for functions). We can have that discussion in the APIView.